dmoj/cptbox/ptproc.cpp

```diff
@@ -67,6 +83,66 @@ int pt_process::set_handler(int syscall, int handler) {
     return 0;
 }
 
+bool pt_process::set_re_fs_read(const char *pattern, int length, char *error,
```

Comment:
Probably want to make this more generic, since we'll likely have the write equivalent of this in the future. Maybe have some struct wrapping all the pointers needed for a single regex?